gulpfile.js

```diff
@@ -0,0 +1,139 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview node module that automate tasks.
+ *
+ * @author barnabasmakonda@gmail.com (Barnabas Makonda)
+ */
+
+var yargs = require('yargs')
```

Comment:
Ah, sorry -- maybe I was unclear. What I meant was, we need a single consistent top-level usage string, otherwise it becomes pretty confusing. For example, if you look at man pages for Unix commands you'll see that each of them has a single synopsis. I think we should do the same here:

```
    $0 <command> [<args>]
```

and the top-level specification is that 'command' is the name of the task you want to run, and each of these tasks comes with their own args. In the case where 'command' is 'build', I think what you have is fine, but I think the 'start_devserver' part needs its own 'command' too, and there should be an error thrown if no 'command' is specified. Then perhaps each of the respective tasks can parse the argv (and raise an error if any of the argv's are wrong or incomplete). So then the only thing this script would be doing on a global level is to parse the command and delegate it to the relevant task/function, and thereafter that task/function is responsible for handling the additional args and the subsequent execution. It also provides a straightforward pattern for extending this script to include additional commands in the future.